### PR TITLE
76 add expected metadata files split by modality

### PR DESCRIPTION
**Pull request by dbirman**
This PR adds a new class ExpectedFiles that tracks whether modalities are expected to have each of the metadata file types. Files can be required, optional, or ignored.

```
>>> from aind_data_schema_models.modalities import ExpectedFiles
>>> ExpectedFiles.ECEPHYS.subject
<FileRequirement.REQUIRED: 1>
>>> ExpectedFiles.ECEPHYS.quality_control
<FileRequirement.OPTIONAL: 0>
```

This also required changing field_handlers to allow passing the field key.

### CODE REVIEW

File: src/aind_data_schema_models/modalities.py [hunk 1]
```diff
@@ -25,3 +27,66 @@ class ModalityModel(BaseName):
...
 Modality.abbreviation_map = {m().abbreviation: m() for m in Modality.ALL}
 Modality.from_abbreviation = lambda x: Modality.abbreviation_map.get(x)
+
+
+class FileRequirement(IntEnum):
+    """Whether a file is required for a specific modality"""
+
+    REQUIRED = 1
+    OPTIONAL = 0
+    IGNORED = -1
```
Comment (saskiad) [inline src/aind_data_schema_models/modalities.py:37]: I think we want a differently label. Ignored sounds equivalent to optional to me - e.g. doesn't matter if it's there or not - whereas what we're saying is files labeled with -1 should actively not be present. 
Not sure the right word though. Barred?

**Review by saskiad — COMMENTED** [review]
I'm just wanting a different label than ignored....

File: src/aind_data_schema_models/modalities.py (hunk 1 above)
Comment (dbirman) [reply src/aind_data_schema_models/modalities.py:37]: How about "Not Applicable" or "Excluded"?

**Review by saskiad — APPROVED** [review]
I like excluded! thanks!

**Pull request merged**